src/seccomp/policy/DefaultPolicy.cc

```diff
@@ -188,6 +188,14 @@ void DefaultPolicy::addFileSystemAccessRules(bool readOnly) {
                 "openat",
                 action::ActionAllow(),
                 (filter::SyscallArg(2) & (O_RDWR | O_WRONLY)) == 0));
+        rules_.emplace_back(SeccompRule(
+                "open",
+                action::ActionErrno(EROFS),
+                (filter::SyscallArg(1) & (O_RDONLY | O_PATH)) == 0));
```

Comment:
I see what you did here - either exactly one of `O_RDONLY`, `O_WRONLY`, `O_RDWR`, `O_PATH` is specified, or the call is invalid and failling back to killing the process and reporting rule violation is fine.
So if it's a valid call, and it's not O_RDONLY and not O_PATH, then it must be O_RDWR or O_WRONLY and we can return `EROFS`.

However,  here's a catch: on Linux, `O_RDONLY` is zero. It's not a real flag.

I think the safest solution would be to do sth like

```c++
for (const auto& mode: {O_RDWR, O_WRONLY}) {
    rules_.emplace_back(SeccompRule(
      "open",
      action::ActionErrno(EROFS),
      (filter::SyscallArg(1) & mode) == 0));
}
```